src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/AbstractAddItemCommand.cs

```diff
@@ -35,7 +35,11 @@ public AbstractAddItemCommand(IPhysicalProjectTree projectTree, IUnconfiguredPro
 
         protected abstract Task OnAddingNodesAsync(IProjectTree nodeToAddTo);
 
-        protected abstract Task OnAddedNodesAsync(ConfiguredProject configuredProject, IEnumerable<IProjectTree> addedNodes, IProjectTree target);
+        protected virtual Task OnAddedNodesAsync(ConfiguredProject configuredProject, IProjectTree target, IEnumerable<IProjectTree> addedNodes, IProjectTree updatedTarget)
+        {
+            // We are not using the updated target so we don't step over the added nodes.
+            return OrderingHelper.TryMoveNodesToTopAsync(configuredProject, addedNodes, target);
```

Comment:
I'm unsure about this behavior; If 10 node were successfully handled, but 1 failed to move - you say that the command wasn't handled and someone else gets to handle it. Why is that?

Reply:
You are referring to that we are ignoring the result of the `Try`?

Reply:
Oh, I was reading this wrong - I assumed it was returning the `Task<bool>` here but its not, its throwing it away.

In that case, why we do spent a bunch of effort capturing whether we successfully added all nodes if we ignore the result?

Reply:
Part of the effort is for tests, where we do look at the result. In this case, we can't test the async version, only the non-async one, `TryMoveNodesToTop`.

Reply:
Because of the lock? Look at the new IProjectAccessor interface I added for this purpose.

Reply:
Unsure if the right stuff in is 15.7 or master.

Reply:
I believe it was for the locks and the instance of the configured project. A plus side to this is separated concerns.

`IProjectAccessor` is identical to what `OrderingHelper.ModifyAsync` is doing, except that I want to read/write the `Project`, which `IProjectAccessor` doesn't have.

Also, how would `IProjectAccessor` be discoverable?